hugr-core/src/hugr/linking.rs

```diff
@@ -0,0 +1,520 @@
+//! Directives and errors relating to linking Hugrs.
+
+use std::{collections::HashMap, fmt::Display};
+
+use itertools::Either;
+
+use crate::{
+    Hugr, HugrView, Node,
+    core::HugrNode,
+    hugr::{HugrMut, internal::HugrMutInternals},
+};
+
+/// Methods for linking Hugrs, i.e. merging the Hugrs and adding edges between old and inserted nodes.
+///
+/// This is done by module-children from the inserted (source) Hugr replacing, or being replaced by,
+/// module-children already in the target Hugr; static edges from the replaced node,
+/// are transferred to come from the replacing node, and the replaced node(/subtree) then deleted.
+pub trait LinkHugr: HugrMut {
```

Comment:
I'm fine with leaving `insert_hugr` and `insert_from_view` as they are for the moment, but could you add a note in their docs pointing users here if they are considering using them?

E.g. say something like "more specialised methods that handle copying multiple module root children and preserving edge connectivity can be found in [`LinkHugr`].